full-service/src/service/transaction_builder.rs

```diff
@@ -518,14 +557,35 @@ impl<FPR: FogPubkeyResolver + 'static> WalletTransactionBuilder<FPR> {
                 )));
             }
 
-            let change_value = input_value - *total_value;
-
+            let mut change_value = input_value - *total_value;
             if change_value > u64::MAX as u128 {
                 return Err(WalletTransactionBuilderError::ChangeLargerThanMaxValue(
                     change_value,
                 ));
             }
 
+            if let Some(fragmentation_settings) = self.fragmentation_settings.get(&token_id) {
+                let num_existing_outputs = *num_outputs_per_token.get(&token_id).unwrap_or(&0);
+                let fragment_destination = if let Some(subaddress_index_to_spend_from) =
+                    self.subaddress_index_to_spend_from
+                {
+                    account.public_address(subaddress_index_to_spend_from)?
+                } else {
+                    reserved_subaddresses.primary_address.clone()
+                };
+
+                add_fragmentation_outputs(
```

Comment:
I debated placing the code of this function here versus having a function with so many mutable arguments, but since this function is already very long I think this is overall better.